### PR TITLE
EC upgrade: close modal bug, preflight polling issue, error message for available updates 

### PR DESCRIPTION
**Pull request by miaawong**
do not run preflight when there are results already add loading state to config next button
better ui for available versions errors

#### What this PR does / why we need it:
https://app.shortcut.com/replicated/story/105489/ui-show-deployment-wizard-when-user-chooses-to-deploy-an-available-version

#### Which issue(s) this PR fixes:

Fixes # [sc-105489] 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: web/src/components/apps/AppVersionHistory.tsx [hunk 1]
```diff
@@ -1906,140 +1917,140 @@ class AppVersionHistory extends Component<Props, State> {
...
-                              <p className="u-fontSize--normal u-fontWeight--medium u-textColor--info">
-                                {this.state.updatesAvailable
-                                  ? "New version available"
-                                  : ""}
-                              </p>
-                              <div className="flex alignItems--center">
+
+                      {!gitopsIsConnected &&
+                        (this.state.updatesAvailable ||
+                          (this.state.availableUpdates &&
```
Comment (miaawong) [inline web/src/components/apps/AppVersionHistory.tsx:1923]: prevents this div from rendering when there aren't any versions 
Version History embedded-cluster-smoke-test-staging-app Admin Console 2024-06-26 at 1 04 04 PM

File: web/src/components/upgrade_service/hooks/getPreflightResult.tsx [hunk 2]
```diff
@@ -190,8 +190,8 @@ function useGetPrelightResults({
       setRefetchCount(0);
     },
     refetchInterval: (preflightCheck: PreflightCheck | undefined) => {
-      if (!preflightCheck) return false;
-      if (!preflightCheck?.pollForUpdates) return false;
+      if (!preflightCheck) return null;
+      if (preflightCheck?.preflightResults.length > 0) return null;
```
Comment (miaawong) [inline web/src/components/upgrade_service/hooks/getPreflightResult.tsx:194]: stops polling once we get results

File: web/src/components/apps/AppVersionHistory.tsx [hunk 3]
```diff
@@ -232,9 +232,11 @@ class AppVersionHistory extends Component<Props, State> {
 
   _mounted: boolean | undefined;
 
-  handleIframeMessage = () => {
-    this.setState({ shouldShowUpgradeServiceModal: false });
-    this.props.navigate(`/app/${this.props.params.slug}`);
+  handleIframeMessage = (event) => {
+    if (event.data.message === "closeUpgradeModal") {
```
Comment (miaawong) [inline web/src/components/apps/AppVersionHistory.tsx:236]: should only run if we post `closeUpgradeModal` message

File: web/src/components/apps/AppVersionHistory.tsx [hunk 4]
```diff
@@ -1470,25 +1473,32 @@ class AppVersionHistory extends Component<Props, State> {
...
+                >
+                  {this.state.upgradeService?.version ===
+                    version.versionLabel && this.state.upgradeService.isLoading
+                    ? "Deploying..."
+                    : "Deploy"}
+                </button>
+              </div>
+              {this.state.upgradeService?.error &&
+                this.state.upgradeService?.version === version.versionLabel && (
+                  <div className="tw-my-4">
```
Comment (miaawong) [inline web/src/components/apps/AppVersionHistory.tsx:1496]: better error message

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**